.github/workflows/credit.yml

```diff
@@ -0,0 +1,43 @@
+name: Contributor credit
+
+on:  # yamllint disable-line rule:truthy
+  # Scheduled actions only run on the main repo branch, which is exactly what we want
+  schedule:
+    # TODO: After making sure it works in `main` for a while, switch to monthly
+    # - cron: '0 0 1 * *'  # first day of the month at midnight
+    - cron: '0 0 * * *'  # every day at midnight
+
+permissions:
+  pull-requests: write
+
+jobs:
+  update_credit:
+    name: Update
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ github.token }}
+      GITHUB_TOKEN: ${{ github.token }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+      - run: pip install pygithub -e .
+      - run: git checkout -b credit
+      - run: python tools/dev/update_credit_json.py
+      - run: python tools/dev/update_credit_rst.py
+      - run: git add -f doc/sphinxext/prs/*.json
+      - run: |
+          git diff && git status --porcelain
+          if [[ $(git status --porcelain) ]]; then
+            echo "dirty=true" >> $GITHUB_OUTPUT
+          fi
+        id: status
+      - name: Create PR
+        run: |
+          set -xeo pipefail
+          git config --global user.email "114827586+autofix-ci[bot]@users.noreply.github.com"
+          git config --global user.name "autofix-ci[bot]"
+          git commit -am "MAINT: Update code credit"
+          gh pr create -B main -H credit --title "MAINT: Update code credit" --body "Created by \"${{ github.workflow }}\" GitHub action." --label "no-changelog-entry-needed"
```

Comment:
This seems like it should in principle work on `main`. I on purpose left out our most recently merged PR #12747 and it got to the stage it would try to open a PR to add it 

https://github.com/mne-tools/mne-python/actions/runs/10275063578/job/28433018563?pr=12774

I set it to try nightly for now. Once we know it works I plan to change it to monthly (seems often enough).